### PR TITLE
Adds a tgui interface for the automated announcement console.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Worked when I tested it. Seemed to respected sanity. Borgs/AIs could still properly use it.

## Why It's Good For The Game

normal html bad or so I've heard.
Please be patient.
I have only a faint idea of what I'm doing.

## Changelog
:cl:
add: tgui interface for the automated announcement console.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/AutomatedAnnouncement.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      <Section
+        title="Departmental Head Announcement"
+        buttons=<Button
+          icon="power-off"
+          selected={data.newheadToggle}
+          onClick={() => {
+            return act('NewheadToggle');
+          }}
+          content={data.newheadToggle ? "On" : "Off"}
+        />>
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/AutomatedAnnouncement.js]:
```suggestion
      <Section
        title="Departmental Head Announcement"
        buttons={(
          <Button
            icon="power-off"
            selected={data.newheadToggle}
            onClick={() => act('NewheadToggle')}
            content={data.newheadToggle ? "On" : "Off"}
          />
        )}>
```

File: tgui/packages/tgui/interfaces/AutomatedAnnouncement.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        <LabeledList>
+          <LabeledList.Item label="Message">
+            <Button content={data.newhead}
+              onClick={() => { return act('NewheadText'); }} />
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/AutomatedAnnouncement.js]:
```suggestion
            <Button
              content={data.newhead}
              onClick={() => act('NewheadText')} />
```

File: tgui/packages/tgui/interfaces/AutomatedAnnouncement.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        </LabeledList>
+      </Section>
+      <Section
+        title="Departmental Head Announcement"
+        buttons=<Button
+          icon="power-off"
+          selected={data.newheadToggle}
+          onClick={() => {
+            return act('NewheadToggle');
+          }}
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/AutomatedAnnouncement.js]: Simpler form is preferred when it's just one function call.

```suggestion
          onClick={() => act('NewheadToggle')}
```

File: tgui/packages/tgui/interfaces/AutomatedAnnouncement.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          selected={data.arrivalToggle}
+          onClick={() => {
+            return act('ArrivalToggle');
+          }}
+          content={data.arrivalToggle ? "On" : "Off"}
+        />>
+        <LabeledList>
+          <LabeledList.Item label="Message">
+            <Button content={data.arrival}
+              onClick={() => { return act('ArrivalText'); }} />
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/AutomatedAnnouncement.js]:
```suggestion
            <Button
              content={data.arrival}
              onClick={() => act('ArrivalText')} />
```

File: tgui/packages/tgui/interfaces/AutomatedAnnouncement.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      <Section
+        title="Arrival Announcement"
+        buttons=<Button
+          icon="power-off"
+          selected={data.arrivalToggle}
+          onClick={() => {
+            return act('ArrivalToggle');
+          }}
+          content={data.arrivalToggle ? "On" : "Off"}
+        />>
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/AutomatedAnnouncement.js]:
```suggestion
      <Section
        title="Arrival Announcement"
        buttons={(
          <Button
            icon="power-off"
            selected={data.arrivalToggle}
            onClick={() => act('ArrivalToggle')}
            content={data.arrivalToggle ? "On" : "Off"}
          />
        )}>
```

**Review by stylemistake — CHANGES_REQUESTED** [review]
Cosmetic stuff, but no technical issues.

**Comment by nemvar** [comment]
Should be all of it. Did it by hand to get a feel for the suggested style. If anything is still off, please point it out.
thanks.

**Comment by Tharcoonvagh** [comment]
Pics?

**Comment by nemvar** [comment]
I switched out the button for those fancy text inputs.
autmated

**Comment by stylemistake** [comment]
Needs rebuilding, also use a `fluid` property on inputs instead of setting the width manually.

**Comment by stylemistake** [comment]
```
ERROR: Permission to nemvar/tgstation.git denied to stylemistake.
fatal: Could not read from remote repository.
```

image

Tick a checkbox to allow contributions from maintainers.

**Pull request closed without merging**